Makefile.rules

```diff
@@ -11,36 +11,36 @@
 ifndef KOS_DEPDIR
 
 %.o: %.c
-	kos-cc $(CFLAGS) -c $< -o $@
+	kos-cc $(CFLAGS) $(CPPFLAGS) -c $< -o $@
 
 %.o: %.cc
-	kos-c++ $(CFLAGS) $(CXXFLAGS) -c $< -o $@
+	kos-c++ $(CXXFLAGS) $(CPPFLAGS) -c $< -o $@
 
 %.o: %.cpp
-	kos-c++ $(CFLAGS) $(CXXFLAGS) -c $< -o $@
+	kos-c++ $(CXXFLAGS) $(CPPFLAGS) -c $< -o $@
 
 %.o: %.m
-	kos-cc $(CFLAGS) -c $< -o $@
+	kos-cc $(CFLAGS) $(CPPFLAGS) -c $< -o $@
 
 %.o: %.mm
-	kos-c++ $(CFLAGS) $(CXXFLAGS) -c $< -o $@
+	kos-c++ $(CFLAGS) $(CXXFLAGS) $(CPPFLAGS) -c $< -o $@
```

Comment:
Should this still have `$(CFLAGS)`. I'd think it would match the rest and that it was accidentally using CFLAGS rather than CPPFLAGS before, but am not sure if obj-c needs both for some reason.

Reply:
ObjC is the one time apparently you do want CFLAGS to trickle over. It's because C++ has shitty C backwards compat, while the spec requires ObjC *strictly* comply with anything standard C can do, so any C flags or anything like that can also be consumed by ObjC.